src/main/java/services/ReportService.java

```diff
@@ -0,0 +1,69 @@
+package services;
+
+import com.google.inject.Inject;
+import dao.PersonsDao;
+import dao.SlotsDao;
+import domain.Persons;
+import domain.Report;
+import domain.Slots;
+import domain.SlotsTimes;
+
+import java.sql.Time;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.Iterator;
+import java.util.List;
+
+public class ReportService {
+
+    private PersonsDao personsDao;
+    private SlotsDao slotsDao;
+
+    @Inject
+    public ReportService(SlotsDao slotsDao, PersonsDao personsDao) {
+        this.slotsDao = slotsDao;
+        this.personsDao = personsDao;
+    }
+
+    public Report getReport(long personId, Date startDate, Date endDate) {
+
+        Long availableCount = 0L;
+        Long fullCount = 0L;
+        Long initCount = 0L;
+
+        Persons person = personsDao.getById(personId);
+
+        List<Slots> slotsList = slotsDao.getForPersonForWeek(personId, startDate, endDate);
+
+        for (Slots slot : slotsList) {
+            SlotsTimes slotTime = slot.getSlot();
+
+            Long slotDuration = (slotTime.getEndTime().getTime() - slotTime.getStartTime().getTime())/60000;
```

Comment:
Extract this magic value into a constant or configuration variable. It would also be worth to extract this calculation into a separate method.